modules/pubmaticBidAdapter.js

```diff
@@ -603,42 +612,85 @@ function _createImpressionObject(bid, conf) {
+              format = [];
+              sizes.forEach(function (size) {
+                format.push({ w: size[0], h: size[1] });
+              });
+              bannerObj.format = format;
+            }
+          } else {
+            utils.logWarn(BIDDER_CODE + 'Error: mediaTypes.banner.size missing for adunit: ' + bid.params.adUnit + '. Ignoring the adunit.');
+          }
+          bannerObj.pos = 0;
+          bannerObj.topframe = utils.inIframe() ? 0 : 1;
+          impObj.banner = bannerObj;
 
-    for (var key in VIDEO_CUSTOM_PARAMS) {
-      if (videoData.hasOwnProperty(key)) {
-        videoObj[key] = _checkParamDataType(key, videoData[key], VIDEO_CUSTOM_PARAMS[key])
-      }
-    }
-    // read playersize and assign to h and w.
-    if (utils.isArray(bid.mediaTypes.video.playerSize[0])) {
-      videoObj.w = bid.mediaTypes.video.playerSize[0][0];
-      videoObj.h = bid.mediaTypes.video.playerSize[0][1];
-    } else if (utils.isNumber(bid.mediaTypes.video.playerSize[0])) {
-      videoObj.w = bid.mediaTypes.video.playerSize[0];
-      videoObj.h = bid.mediaTypes.video.playerSize[1];
-    }
-    if (bid.params.video.hasOwnProperty('skippable')) {
-      videoObj.ext = {
-        'video_skippable': bid.params.video.skippable ? 1 : 0
+          break;
+        case NATIVE:
+          nativeObj['request'] = JSON.stringify(_createNativeRequest(bid.nativeParams));
+          if (!isInvalidNativeRequest) {
+            impObj.native = nativeObj;
+          } else {
+            utils.logWarn(BIDDER_CODE + 'Error: Error in Native adunit ' + bid.params.adUnit + '. Ignoring the adunit. Refer to ' + PREBID_NATIVE_HELP_LINK + ' for more details.');
+          }
+
+          break;
+        case VIDEO:
+          var videoData = bid.params.video;
+          if (videoData !== undefined) {
+            for (var key in VIDEO_CUSTOM_PARAMS) {
+              if (videoData.hasOwnProperty(key)) {
+                videoObj[key] = _checkParamDataType(key, videoData[key], VIDEO_CUSTOM_PARAMS[key]);
+              }
+            }
+            // read playersize and assign to h and w.
+            if (utils.isArray(bid.mediaTypes.video.playerSize[0])) {
+              videoObj.w = bid.mediaTypes.video.playerSize[0][0];
+              videoObj.h = bid.mediaTypes.video.playerSize[0][1];
+            } else if (utils.isNumber(bid.mediaTypes.video.playerSize[0])) {
+              videoObj.w = bid.mediaTypes.video.playerSize[0];
+              videoObj.h = bid.mediaTypes.video.playerSize[1];
+            }
+            if (bid.params.video.hasOwnProperty('skippable')) {
+              videoObj.ext = {
+                'video_skippable': bid.params.video.skippable ? 1 : 0
+              };
+            }
+
+            impObj.video = videoObj;
+          } else {
+            utils.logWarn(BIDDER_CODE + 'Error: Video config params missing for adunit: ' + bid.params.adUnit + ' with mediaType set as video. Ignoring the adunit.');
```

Comment:
here we are ignoring video part of the adunit, not the whole adUnit, we need to correct the message here, also for other mediaTypes.